projects/step-frontend/src/lib/modules/execution/components/schedule-overview/cross-execution-dashboard/cross-execution-dashboard.component.ts

```diff
@@ -24,6 +24,29 @@ export class CrossExecutionDashboardComponent implements OnInit {
 
   protected tabs: Tab<string>[] = [this.createTab('report', 'Report'), this.createTab('performance', 'Performance')];
 
+  viewTitle = computed(() => {
+    if (this._state.viewType() === 'task') {
+      let task = this._state.task();
+      if (task === undefined) {
+        return 'Loading...';
+      } else if (task === null) {
+        return 'Task name (deleted)';
+      } else {
+        return task?.attributes?.['name'];
+      }
+    } else {
+      // plan
+      const plan = this._state.plan();
+      if (plan === undefined) {
+        return 'Loading...';
+      } else if (plan === null) {
+        return 'Plan name (deleted)';
+      } else {
+        return plan.attributes?.['name'];
+      }
+    }
+  });
```

Comment:
Not sure I understand why we are adding "name" here. This is shown in the top of the view, correct? In that case I would write "Deleted task" / "Deleted plan"

Also the code could be improved as all except for the deleted part is the same (please check as I wrote it in the comment field untested):

> const isTask = this._state.viewType() === 'task';
> let entity = isTask ? this._state.task() : this._state.plan();
>       if (entity === undefined) {
>         return 'Loading...';
>       } else if (entity === null) {
>         return isTask ? "Deleted task" : "Deleted plan";
>       } else {
>         return entity?.attributes?.['name'];
>       }
>